packages/slice-machine/components/ScreenshotChangesModal/Views/index.tsx

```diff
@@ -58,16 +72,49 @@ function VariationScreenshot({
     onHandleFile: generateSliceCustomScreenshot,
   });
 
+  const sliceView = useMemo(
```

Comment:
interesting, didn't think useMemo could do the equivalent of a useState + useEffect.

Reply:
I didn't actually write this, took it from the slice builder. Your comment made me read into it though and it's quite interesting: https://stackoverflow.com/questions/56028913/usememo-vs-useeffect-usestate

Reply:
Yeah super interesting as well :), seems to do the same as useCallback so I wonder what would be the difference between these two.